### ISSUE TITLE
Frequent LSP error messages

### ISSUE THREAD

**Issue by askfiy** [jose-elias-alvarez/null-ls.nvim#865]
I've only been using `null-ls` for half a month, but I'm running into problems that never happened before.

```
LSP[id=2] client has shut down after sending the message
```

This error message occurs frequently, and I'm sure it's generated after the `vim.lsp.start_client()` method is run.

For this, I searched for occurrences of `vim.lsp.start_client()` in all my plugins.

Apparently, this method is only called in `null-ls` and `lspconfig`.

But this method is called only once in `lspconfig`, but many times in `null-ls`.

So I have absolutely reason to suspect that this error message is caused by `null-ls`.

It only happens in `Lua` files, sometimes it appears when I switch to another `buffer`, sometimes it also appears when I open `Telescope` to search.

This is really annoying, this error message doesn't help me anything, it just annoys me...

**Comment by jose-elias-alvarez** [comment]
Your analysis is not accurate - while it may occur only once in nvim-lspconfig's code, it calls the relevant method once per client. Moreover, this specific error message comes from [this line](https://github.com/neovim/neovim/blob/8fba428bc6f36ae038a9286517e15b33257a1359/runtime/lua/vim/lsp/handlers.lua#L164), which refers to a method that null-ls doesn't even support (I have seen the same message from lua-language-server).

Because you deleted the template, this issue is not actionable, so I'm closing it. Please open another issue if you are able to provide a reproduction isolated to null-ls.

**Comment by askfiy** [comment]
Thank you very much, do you have any good suggestions for this?

**Comment by jose-elias-alvarez** [comment]
Do you mean to investigate whether null-ls is involved? We have a minimal init.lua file in the issue template that should provide a good starting point. The fact that you're only seeing this with Lua files indicates to me that it's related to lua-language-server, though, since we don't have any special handling for Lua files, so I would personally start there. 

Also, if all you care about is ignoring the error message (I'm pretty sure it's harmless) you can also just filter it out:

```lua
local notify = vim.notify
vim.notify = function(msg, ...)
    if msg:match("client has shut down after sending the message") then
        return
    end

    notify(msg, ...)
end
```

**Comment by askfiy** [comment]
Thank you, You're such a warm-hearted person
